tools/AzureRM.BootStrapper/AzureRM.BootStrapper.Tests.ps1

```diff
@@ -0,0 +1,231 @@
+           It "Should return installed version" {
+                Get-AzureRmModule -Profile 'Profile1' -Module 'Module1' | Should Be "1.0"
+                Assert-VerifiableMocks
+            }
+        }
+        
+        Context "Module is not installed" {
+            Mock Get-Module -Verifiable {}
+            It "Should return null" {
+                Get-AzureRmModule -Profile 'Profile1' -Module 'Module1' | Should be $null
+                Assert-VerifiableMocks
+            }
+        }
+    }
+}
+
+Describe "Get-AzureRmProfile" {
+    InModuleScope AzureRM.Bootstrapper {
+        Mock Get-AzProfile -Verifiable { ($global:testProfileMap | ConvertFrom-Json) }
+
+        Context "With ListAvailable Switch" {
+            It "Should return available profiles" {
+                $Result = (Get-AzureRmProfile -ListAvailable)
+                $Result.Length | Should Be 14
+                $Result[0] | Should Be "Profile: Profile1"
+                Assert-VerifiableMocks
+            }
+        }
+
+        Context "With ListAvailable and Force Switches" {
+            It "Should return available profiles" {
+                $Result = (Get-AzureRmProfile -ListAvailable -Force)
+                $Result.Length | Should Be 14
+                $Result[0] | Should Be "Profile: Profile1"
+                Assert-VerifiableMocks
+            }
+        }
+
+        Context "Without ListAvailable Switch" {
+            $RollupModule = 'Module1'
+            Mock Get-AzureRmModule -Verifiable { "1.0"} -ParameterFilter {$Profile -eq "Profile1" -and $Module -eq "Module1"}
+            It "Returns installed Profile" {
+                (Get-AzureRmProfile) | Should Be "Profile1"
+                Assert-VerifiableMocks
+            }
+        }
+
+        Context "No profiles installed" {
+            $RollupModule = 'Module1'
+            Mock Get-AzureRmModule -Verifiable {} -ParameterFilter {$Profile -eq "Profile1" -and $Module -eq "Module1"}
+            It "Returns null" {
+                (Get-AzureRmProfile) | Should Be $null
+                Assert-VerifiableMocks
+            }
+        }
+    }
+}
+
+Describe "Use-AzureRmProfile" {
+    InModuleScope AzureRM.Bootstrapper {
+        $RollupModule = 'Module1'
+        Mock Get-AzProfile -Verifiable { ($global:testProfileMap | ConvertFrom-Json) }
+        Mock Install-Module { "Installing module..."}
+        Mock Import-Module -Verifiable { "Importing Module..."}
+
+        Context "Modules not installed" {
+            Mock Get-AzureRmModule -Verifiable {} -ParameterFilter {$Profile -eq "Profile1" -and $Module -eq "Module1"}
+            It "Should install modules" {
+                $Result = (Use-AzureRmProfile -Profile 'Profile1' -Force)
+                $Result.Length | Should Be 2
+                $Result[0] | Should Be "Installing module..."
+                $Result[1] | Should Be "Importing Module..."
```

Comment:
It would be nice to verify the version of the moduel imprted, and to do this for modules with > 1 version installed